content/guides/token-extensions/transfer-hook.md

```diff
@@ -337,6 +337,54 @@ Running tests...
   4 passing (2s)
 ```
 
+<Callout type="warning">
+
+Since here we are increasing a counter whenever the token is transferred we need
+to make sure that the transfer hook instruction can only be called during a
+transfer, otherwise someone could just call the transfer hook instruction
+directly and mess up our counter. This is a check you should add to any of your
+transfer hooks.
+
+</Callout>
+
+You can add the check like this:
+
+```rust
+fn check_is_transferring(ctx: &Context<TransferHook>) -> Result<()> {
```

Comment:
Since assertions are a common practice, let's update the name of this function to use an "assertion" type name?

```suggestion
fn assert_is_transferring(ctx: &Context<TransferHook>) -> Result<()> {
```